src/terrasoApi/account/accountService.ts

```diff
@@ -55,20 +52,12 @@ export const getAuthURLs = () =>
     })
   );
 
-type UserQuery = {
-  users: {
-    edges: {
-      node: UserFragment & UserPreferencesFragment;
-    }[];
-  };
-};
-
 export const fetchProfile = async (
-  params: any,
+  params: null,
   currentUser: { email: string } | null
 ) => {
-  const query = `
-    query user($email: String!){
+  const query = graphql(`
```

Comment:
Question: What is the `graphql` function doing? is doing a validation? more than that? or something different?

Reply:
ok here is a full explanation of how the code generation works:

- by parsing `src/terrasoApi/schema.graphql`, the code gen understand which types exist in our schema, which queries & mutations are allowed, and what the types of their results are
- each _fragment_ tagged with a `/* GraphQL */` comment is validated to ensure all the fields it references exist, and a corresponding fragment type is created
- each query or mutation tagged with a ``graphql(` ... `)`` call is validated to ensure all fields and inputs exist, and an entry is created in a generated dictionary where the keys are the literal strings in the `graphql` call, and the values are the query strings with fragment spreads expanded (at code generation time) from fragments ingested via `/* GraphQL */` tags. the expanded query strings are also wrapped with the type information of the inputs and returned fields (which our `requestGraphQL` function uses to provide static typing for input and result JSON objects).
- a function `graphql` is generated which:
  - at compile time, knows what the corresponding input and result types are based on the literal string argument
  - at runtime, simply looks up its argument in the generated dictionary

if you run `npm run generate` in a local project, you can see the code which gets generated under `src/terrasoApi/gql/...`

some implications:
- you no longer need to write the fragment spread name twice. i think there is a way to set up your local tooling so that jump to definition works on the graphql string itself
- the graphql name of fragments and queries (the name you give them within the graphql string) must now be globally unique. (this is a bit annoying but i think overall worth it).
- this doesn't affect existing code, which can still operate as normal.
- you can't build queries programmatically via string interpolation using this method, the only way to re-use query code is via fragments and inputs. there are features available for type-safe programmatic manipulation of queries if we want to do that, but it's more complicated and wasn't needed for this PR so i left it out
- performance: i don't think significant difference, we aren't doing runtime string interpolation but we are doing a runtime dictionary lookup. i guess it's a bit faster, but i doubt this is a bottleneck either way

Reply:
The implications look managable for me, the big one related to the development environment and being able to just add changes without running a separate script everytime I think is important IMO. The one related to string interpolation I think is fine, ideally we should pass values through variables that as far as I see this is still supported.

Reply:
see https://github.com/techmatters/terraso-web-client/pull/934#issuecomment-1552260099 re dev environment